ocv_module/open_model_zoo/gen.py

```diff
@@ -0,0 +1,164 @@
+    else:
+        print('Unexpected source instance: %s', type(source))
+        sys.exit(1)
+
+    return url, sha, name
+
+
+# List of models which have versions (mostly DLDT models). For such kind of models
+# with the highest version we will add short names. In example, for
+# "face_detection_retail_0004" and "face_detection_retail_0005" method with name
+# "face_detection_retail" returns result of "face_detection_retail_0005". The same
+# approach for precisions: "face_detection_retail_fp16" for "face_detection_retail_0005_fp16"
+versionedNames = {}
+
+def registerVersionedName(fullName, precision=''):
+    global versionedNames
+
+    matches = re.search('(.+)_(\d{4})$', fullName)
+    if matches:
+        shortName = matches.group(1) + ('_%s' % precision if precision else '')
+        version = matches.group(2)
+        if not shortName in versionedNames or int(version) > int(versionedNames[shortName][0]):
+            originName = fullName + ('_%s' % precision if precision else '')
+            versionedNames[shortName] = [version, originName]
+
+
+def generate(topology, topologyName, topologies_hdr, py_impl, cpp_impl):
+    name = topologyName.replace('-', '_').replace('.', '_')
+
+    # DLDT models come with multiple files foe different precision
+    files = topology['files']
+    assert(len(files) > 0), topologyName
+    if len(files) > 2:
+        assert(topology['framework'] == 'dldt'), topologyName
+        assert(len(files) % 2 == 0), topologyName
+
+        for i in range(len(files) / 2):
+            subTopology = topology.copy()
+            subTopologyName = topologyName
+            subTopology['files'] = [files[i * 2], files[i * 2 + 1]]
+            # Detect precision by the first file
+            precision = subTopology['files'][0]['name']
+            precision = precision[:precision.find('/')].lower()
+            if precision != 'fp32':  # Keep origin name for FP32
+                subTopologyName += '_' + precision
+                registerVersionedName(name, precision)
+            generate(subTopology, subTopologyName, topologies_hdr, py_impl, cpp_impl)
+        return
+
+
+    registerVersionedName(name)
+
+    config = {}
+
+    config['framework'] = topology['framework']
+
+    config['topology_name'] = name
+    if 'model_optimizer_args' in topology:
+        config['model_optimizer_args'] = ' '.join(topology['model_optimizer_args'])
+
+    fileURL, fileSHA, fileName = getSource(files[0])
+    if fileName.endswith('.tar.gz') or fileName.endswith('.zip'):
+        config['archive_url'], config['archive_sha256'], config['archive_name'] = fileURL, fileSHA, fileName
+    else:
+        config['config_url'], config['config_sha256'], config['config_path'] = fileURL, fileSHA, fileName
+        if len(files) > 1:
+            config['model_url'], config['model_sha256'], config['model_path'] = getSource(files[1])
+
+    s = ', '.join(['{"%s", "%s"}' % (key, value) for key, value in config.items()])
```

Comment:
No need to cast into a list.
```suggestion
    s = ', '.join('{"%s", "%s"}' % (key, value) for key, value in config.items())
```